test/e2e/apimachinery/etcd_failure.go

```diff
@@ -84,7 +84,7 @@ func etcdFailTest(ctx context.Context, f *framework.Framework, failCommand, fixC
 
 	checkExistingRCRecovers(ctx, f)
 
-	apps.TestReplicationControllerServeImageOrFail(ctx, f, "basic", framework.ServeHostnameImage)
```

Comment:
why this needs to be changed? is this not already doing it?

https://github.com/kubernetes/kubernetes/blob/9c9a0e3ba57cb103d651e77f1b9283673c4d0cd7/test/e2e/framework/util.go#L141-L142

Reply:
because global variable initialization in executed before flag parsing, so `ServeHostnameImage` will store the default image, instead of the image specified in flag.